protocols/imap/src/test/java/org/apache/james/imap/processor/base/UidMsnMapperTest.java

```diff
@@ -214,4 +217,83 @@ public void removeShouldKeepAValidMappingWhenDeletingMiddle() {
                 3, messageUid4));
     }
 
+    @Test
+    public void addAndRemoveShouldHaveGoodConcurrentBehaviorWellWhenMixed() throws Exception {
+        final int initialCount = 1000;
+        for (int i = 1; i <= initialCount; i++) {
+            testee.addUid(MessageUid.of(i));
+        }
+
+        int threadCount = 2;
+        ConcurrentTestRunner concurrentTestRunner = new ConcurrentTestRunner(threadCount, initialCount,
+            new ConcurrentTestRunner.BiConsumer() {
+                @Override
+                public void consume(int threadNumber, int step) throws Exception {
+                    if (threadNumber == 0) {
+                        testee.remove(MessageUid.of(step + 1));
```

Comment:
I don't know a lot `ConcurrentTestRunner` does this consume will be executed `initialCount` if yes I think the name is not really good.